.github/workflows/test.yml

```diff
@@ -100,23 +99,35 @@ jobs:
       fail-fast: false
       matrix:
         os: [windows-latest, ubuntu-latest, macos-latest]
-        release: [true, false]
+        profile: [debug, release]
         toolchain: [stable]
         include:
           - os: ubuntu-latest
-            release: false
+            profile: debug
             toolchain: nightly
     env:
       RUSTFLAGS: -D warnings
       RUST_BACKTRACE: 1
       CARGO_FLAGS: --verbose --locked
+      CARGO_INCREMENTAL: 0
     steps:
-      - name: Setup release environment
-        if: ${{ matrix.release }}
+      - name: Set up environment variables and cargo config
         shell: bash
         run: |
-          echo "RUSTFLAGS=$RUSTFLAGS -C lto=no" >> $GITHUB_ENV
-          echo "CARGO_FLAGS=$CARGO_FLAGS --release" >> $GITHUB_ENV
+          if [ '${{ matrix.profile }}' == release ]; then
+            echo "CARGO_FLAGS=$CARGO_FLAGS --release" >> $GITHUB_ENV
+          fi
+          case '${{ matrix.os }}' in
+            ubuntu-latest)  features=systemd,gelf ;;
+            *)              features=gelf ;;
+          esac
+          echo "JORMUNGANDR_FEATURES=$features" >> $GITHUB_ENV
+
+          mkdir .cargo
+          cat > .cargo/config.toml <<EOF
```

Comment:
We may just fill in `RUSTFLAGS` instead of putting everything into a file.

Reply:
The failed build on 6469cf99538b119eb1767114cd75237b2cf39955 is a demonstration why we can't: `RUSTFLAGS` are also used when building things like proc macros and build scripts, and some of these targets fail to compile when told to use LTO.

A Cargo config file is also a neat way to replace conditionals in scripts or step descriptors: the `[profile.release]` section only applies to release builds. The same trick is used in the release workflow to enable LTO precisely for the target binaries.